src/mpyl/project.py

```diff
@@ -134,12 +129,12 @@ class StageSpecificProperty(Generic[T]):
     deploy: Optional[T]
     postdeploy: Optional[T]
 
-    def for_stage(self, stage: Stage) -> Optional[T]:
-        if stage == Stage.BUILD:
+    def for_stage(self, stage: str) -> Optional[T]:
+        if stage == "build":
             return self.build
-        if stage == Stage.TEST:
+        if stage == "test":
             return self.test
-        if stage == Stage.DEPLOY:
+        if stage == "deploy":
             return self.deploy
         return self.postdeploy
```

Comment:
Hm these are still hardcoded.. But I don't think we can do much about it, maybe if we refactor this class to use a dictionary and then extract the properties by dict.get(stage)

Reply:
That would be possible, but the biggest issue I see is that we can't make the schema dynamic.